awx/api/tests/functional/test_api_generics.py

```diff
@@ -0,0 +1,26 @@
+import pytest
+
+# We have to tell flake 8 to ignore these since they are fixtures we need to import
+from awx.conf.tests.functional.conftest import api_request, admin  # noqa: F401; pylint: disable=unused-variable
```

Comment:
The linter isn't happy about the unused `api_request`, and maybe I kind of agree. Why do you think they should be imported? Pytest handles fixtures for you, something in the fixture list doesn't need an import.

Reply:
I could be off base here but I thought the conftest was only picked up if its in the same directory as the test. To avoid code duplication I am pulling in fixtures from awx/conf/tests whereas this test is in awx/api/tests. So without the import `py.test awx/api/tests/functional/test_api_generics.py` vields:
```
_______________________________________________________________ ERROR at setup of test_custom_400_error_log ________________________________________________________________
file /awx_devel/awx/api/tests/functional/test_api_generics.py, line 22
  @pytest.mark.django_db
  def test_custom_400_error_log(caplog, api_request, settings):
E       fixture 'api_request' not found
>       available fixtures: _dj_autoclear_mailbox, _django_clear_site_cache, _django_db_helper, _django_db_marker, _django_set_urlconf, _django_setup_unittest, _fail_for_invalid_template_variable, _live_server_helper, _template_string_if_invalid_marker, admin_client, admin_user, async_client, async_rf, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, client, cov, db, django_assert_max_num_queries, django_assert_num_queries, django_capture_on_commit_callbacks, django_db_blocker, django_db_createdb, django_db_keepdb, django_db_modify_db_settings, django_db_modify_db_settings_parallel_suffix, django_db_modify_db_settings_tox_suffix, django_db_modify_db_settings_xdist_suffix, django_db_reset_sequences, django_db_serialized_rollback, django_db_setup, django_db_use_migrations, django_mail_dnsname, django_mail_patch_dns, django_test_environment, django_user_model, django_username_field, doctest_namespace, live_server, mailoutbox, mock, mocker, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, rf, settings, testrun_uid, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory, transactional_db, worker_id
>       use 'pytest --fixtures [testpath]' for help on them.
```

Reply:
Resolved by moving tests.